File: internal/model/ws/message.go
package ws

// outbound
type MessageDeletedPayload struct {
	ConversationID int `json:"conversation_id"`
	MessageID      int `json:"message_id"`
}

type MessageSeenPayload struct {
	UserID         int `json:"user_id"`
	ConversationID int `json:"conversation_id"`
	MessageID      int `json:"message_id"`
}

type MessageEditedPayload struct {
	ConversationID int    `json:"conversation_id"`
	MessageID      int    `json:"message_id"`
	Body           string `json:"body"`
}

// inbound
type MessageReadPayload struct {
	ConversationID int `json:"conversation_id"`
	MessageID      int `json:"message_id"`
}

File: frontend/src/types/ws/message.ts
export type MessageReadPayload = {
    conversation_id: number
    message_id: number
}

export type MessageEditedPayload = {
    conversation_id: number
    message_id: number
    body: string
}

export type MessageSeenPayload = {
    user_id: number
    conversation_id: number
    message_id: number
}

export type MessageDeletedPayload = {
    conversation_id: number
    message_id: number
}

